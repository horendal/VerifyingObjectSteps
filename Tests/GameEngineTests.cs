using ApprovalTests;
using ApprovalTests.Reporters;
using HangmanGame;
using System.Text;
using Xunit;

namespace Tests
{
    public class GameEngineTests
    {
        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public void GameState()
        {
            var sb = new StringBuilder();

            var sut = new GameEngine("Pluralsight");

            sut.Guess('x');
            sut.Guess('p');
            sut.Guess('l');

            sb.AppendLine(sut.ToString());

            Approvals.Verify(sb.ToString());
        }
    }
}
