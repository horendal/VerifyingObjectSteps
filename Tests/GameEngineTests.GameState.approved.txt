Guessed letters: XPL
Guesses remaining: 9
PLURALSIGHT
PL---L-----
